AdvLoggerPkg/Library/AdvancedLoggerAccessLib/AdvancedLoggerAccessLib.c

```diff
@@ -136,6 +168,40 @@ FormatPhasePrefix (
   return (UINT16)PhaseStringLen;
 }
 
+/**
+  FormatDebugLevelPrefix
+
+  Adds a debug level indicator to the message being returned.  If debug level is recognized and specified,
+  returns the debug_level prefix in from the AdvMsgEntryPrefix, otherwise raw debug level value is returned.
+
+  @param  MessageBuffer
+  @param  MessageBufferSize
+  @param  Phase
+
+  @retval Number of characters printed
+*/
+STATIC
+UINT16
+FormatDebugLevelPrefix (
+  IN CHAR8   *MessageBuffer,
+  IN UINTN   MessageBufferSize,
+  IN UINT32  DebugLevel
+  )
+{
+  UINTN  DebugLevelStringLen;
+

```

Comment:
Can you please validate the input buffers before you use them? and MessageBufferSize. If a size zero buffer is passed in, there is no need to do anything else.

Reply:
Done, thanks Aaron!